### PR TITLE
Add kableone function to convert to markdown

### LINKED ISSUE #24: output with R markdown
Hi,

thank you very much such a great package, it is very handy, 
I am trying to output the result from createTableOne to doc file, but neither r markdown not html works, and Copy/Paste from the console will mixed up the format. How do you output the table to other document?

### PR DESCRIPTION
**Pull request by malcolmbarrett**
Closes #24. 

This PR adds the `kableone()` function that converts a tableone object to markdown. It's extremely simple. It takes advantage of the fact that the print methods for this objects invisibly output a `data.frame`.  It silently calls the print method and then passes the `data.frame` to `knitr::kable()`

### CODE REVIEW

**Comment by malcolmbarrett** [comment]
If you decide to merge this, it may also be good to add as.data.frame methods for the table classes if one just wants it stored in a df for manipulation

**Comment by kaz-yos** [comment]
Thanks!

**Comment by santarabantoosoo** [comment]
this is perfect. However, print.tableone offers many useful arguments that I can't use with kableone. 

In addition, is there a possibility that the output is a dataframe not a kable output, as I want to use the output with flextable

**Comment by malcolmbarrett** [comment]
@kaz-yos and I talked about this a bit. The problem is that both `kable()` and `print.tableone()` use `...`. 

Maybe I should PR a set of arguments like `print_args = list()` & `kable_args = list()`? What do you think, Kaz?

**Comment by santarabantoosoo** [comment]
I have written a function that converts the output into a flextable ..

for ex: 
foo <- print(tableOne, showAllLevels = TRUE, missing = T, minMax = TRUE)

a <- data.frame(foo)

setDT(a, keep.rownames = TRUE)[]

a$rn <- str_replace_all(a$rn, "^X.*", "")

a$rn <- str_replace_all(a$rn, "\\.", " ")

colnames(a)[1] <- "category"

rmdtable <- function(df){
 
  bes <- autofit(theme_vanilla(flextable(df)))

bes <- bg(bes, bg = "blue", part = "header")
bes <- color(bes, color = "white", part = "header")

return(bes) 
}

rmdtable(a)


The output table is more than perfect using Rmarkdown. The only problem is that the print statement produces an output in Rmarkdown as well, which I have to delete manually

**Comment by mikedolanfliss** [comment]
I'm just a lurker, but a big fan of the packge. @malcolmbarrett, I've seen options lists used elsewhere in R to good effect, like the way RCurl passes [curlOptions](https://www.rdocumentation.org/packages/RCurl/versions/1.95-4.12/topics/curlOptions). That would make sense to me.

**Comment by malcolmbarrett** [comment]
I filed a new issue for this at #47.

@santarabantoosoo I haven't run your code, but I suspect you need `capture.output()` `capture.output(foo <- print(tableOne, showAllLevels = TRUE, missing = T, minMax = TRUE))`, although my proposal in the issue would eliminate the need for that

**Comment by santarabantoosoo** [comment]
I tried capture.output but doesn't seem to work ..

I even tried adding the print statement in a separate chunk, with include = FALSE, but again nothing

**Comment by malcolmbarrett** [comment]
how about adding `printToggle = FALSE`? to the print function call?

**Comment by santarabantoosoo** [comment]
I tried it :(   and again nothing :(

**Comment by altonrus** [comment]
Hi @santarabantoosoo, I used your workaround and got it to knit to word without printing by combining the print(tableOne) and data.frame() into one line of code.

Instead of:
foo <- print(tableOne, showAllLevels = TRUE, missing = T, minMax = TRUE)
a <- data.frame(foo)

I did 
a <- data.frame(print(tableOne, showAllLevels = TRUE, missing = T, minMax = TRUE))

**Comment by 11mkapadia** [comment]
hard to follow the multiple issues, but was there ever a solution to get  kable() and print.tableone() to work together?

**Comment by 11mkapadia** [comment]
never mind just realized this works:

print(tab1, varLabels = TRUE, explain = FALSE, printToggle = FALSE) %>% as.data.frame() %>%  kable()  %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
